value.go

```diff
@@ -289,6 +289,20 @@ func tryGetMappedSliceValue[T any](val T, isArray bool, sliceLength int) (mappin
 	if elementLogicType.Ptr == nil {
 		return elementLogicType, mapping.Value{}, unsupportedTypeError(reflect.TypeOf(val).Name())
 	}
+
+	// Validate that all non-null elements have the same type to prevent mixed-type crashes
+	expectedType := mapping.GetTypeId(elementLogicType)
+	for i, lt := range childLogicTypes {
+		if lt.Ptr != nil {
+			currentType := mapping.GetTypeId(lt)
```

Comment:
`GetTypeId` is only the top-level type - thus, I think we need to make sure we compare types recursively here (using their logical types to get child-type IDs). The following test still fails:

```Go
	mixedSlice := [][]any{{"hello this is text"}, {2}}
	_, err = db.Query("FROM mixed_slice_test WHERE foo IN ?", mixedSlice)
```